solidity/dashboard/src/components/Web3ContextProvider.jsx

```diff
@@ -48,10 +49,24 @@ class Web3ContextProvider extends React.Component {
       )
       yourAddress = accounts[0]
 
+      if (
+        this.state.connector?.name === WALLETS.EXPLORER_MODE.name &&
+        connector.name !== WALLETS.EXPLORER_MODE.name
+      ) {
+        if (
+          this.state.yourAddress &&
+          !isSameEthAddress(this.state.yourAddress, yourAddress)
+        ) {
```

Comment:
Nested ifs. We can simplify it to one. If a condition is too long let's extract it to a separate function.